src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,309 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/em/distribution/MollerEnergyDistribution.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/math/Algorithms.hh"
+#include "celeritas/em/data/WentzelData.hh"
+#include "celeritas/em/xs/MottXsCalculator.hh"
+#include "celeritas/em/xs/WentzelXsCalculator.hh"
+#include "celeritas/mat/IsotopeView.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+/*
+ * References:
+ *      Fern - Fernandez-Varea 1992
+ *      PRM - Geant4 Physics Reference Manual Release 11.1
+ */
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Helper class for \c WentzelInteractor .
+ *
+ * Samples the scattering direction for the Wentzel model.
+ */
+class WentzelDistribution
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using MomentumSq = units::MevMomentumSq;
+    using Energy = units::MevEnergy;
+    using Mass = units::MevMass;
+    //!@}
+
+  public:
+    //! Construct with state and date from WentzelInteractor
+    inline CELER_FUNCTION
+    WentzelDistribution(real_type inc_energy,
+                        real_type inc_mass,
```

Comment:
My understanding was that `ParticleTrackView::mass()` was preferred over `constants::electron_mass` where available, and that I shouldn't necessarily assume they're the same.

For `element_data` I'm passing a reference to that struct because otherwise I'd have to pass both the material and `ElementComponentId` and repeat the same lookup in the `WentzelDistribution`. This just let's me pass one object instead of two.

Reply:
Sorry, yes I have `WentzelData::electron_mass` which I put in when I first started and haven't really noticed since. However I'd be now tempted to remove it since it's cached data that's readily passed by parameters.